C#/C#.sublime-syntax

```diff
@@ -1361,29 +1363,32 @@ contexts:
             2: punctuation.separator.cs
             3: punctuation.section.brackets.end.cs
             4: keyword.operator.pointer.cs
-    - match: \(
-      scope: punctuation.section.group.begin.cs
+    - match: (?=\()
       branch_point: tuple_or_group
       branch:
-        - group
-        - tuple
+        - - match: \(
```

Comment:
I'd vote to avoid such anonymous constructs whenever possible.

We can easily use `tuple` vs. `tuple-content` or `tuple-body`. The first one containing the opening bracket and the latter ones could be created by renaming the current `tuple`  context.

Also `prototypes` could be excluded here as the `(?=\()` lookahead already implies the next token must be matched as opening parenthesis. So no any injected/prepended pattern would be nonsense. (I have templating languages in mind, which like to inject their variables to interpolate with syntaxes)

Reply:
I like your level of thinking - I wonder if such an optimization to skip prototypes in such a case could be applied automatically by the syntax engine. If the parent context doesn't skip the prototype, the match is a lookahead and the child context uses the same regex pattern from inside the lookahead: then no need to apply the prototype. I guess it could get complicated where a context is referenced in multiple places, so maybe we can somehow work on a linter to suggest such changes for a potential small perf boost? :D

Reply:
Issues like the one resolved by https://github.com/sublimehq/Packages/pull/2910 teach you this.